### ISSUE TITLE
create link directly to a specific line in another file (not just the first line of the file)

### ISSUE THREAD

**Issue by seelaman** [vimwiki/vimwiki#890]
Prior to submitting a new issue make sure to complete these steps:

- [x] Checkout the `dev` branch and confirm the issue is present there as well.
      The `dev` branch contains fixes that may not have been merged to `master` yet.
- [x] Post the syntax you are using (default/mediawiki/markdown) **and** your vimwiki settings from your `.vimrc`
let g:vimwiki_list = [{'path': '~/notes/',
                      \ 'syntax': 'default', 'ext': '.md'}]
- [x] Provide a detailed description of the problem including **steps to reproduce the issue**.
not really an issue (love VimWiki!), more of a feature request:

I wrote a script that creates bidirectional links, an idea taken from RoamResearch:

For example the following line in a file called BuJo.md
- [ ] [[VimWiki]] write a github issue for backlinks

Will get picked up by the script which will add the following line to VimWiki.md
BuJo.md:609 - [ ] [[VimWiki]] write a github issue for backlinks

This is nice because now I have all my tasks related to a "tag", in this case VimWiki, grouped in VimWiki.md.

The question is: Can the first part (BuJo.md:609) be turned  into a link that will jump directly to line 609 in BuJo.md? I know the link the file itself is no problem but it would be convenient to just hit enter and get right to the spot where the "tag" was used in BuJo.md.

- [x] Include the output of `:VimwikiShowVersion`.
Stable version: '2.4.1'

**Comment by Nudin** [comment]
I'm not sure if I understand what you want to achieve. Linking to a line number it not (yet) possible. You can however link to anchors see `:h vimwiki-anchors` for description but in short, you can link to:
- every heading
- every tag
- every blow text
Maybe this already covers your needs?

**Comment by tinmarino** [comment]
He wants to create a custom jumper to jump wherever he wants: could be ccp source code. It can be great to keep some hand made indexes.

@seelaman : I like the feature, feel free to make a PR, there will be some discussion on the syntax thought, look at the anchor separator witch is `#` and if followed by `:` it goes to a specific line and by '%' does a search with the rest. If you make those '%' and ':' some variable then we can accept the PR and the discussion comes later.

Have a look at base.vim#follow_link and your line or regex would enter in the url part (so cannot contains `[]()`)

**Comment by seelaman** [comment]
> He wants to create a custom jumper to jump wherever he wants
Exactly, I know it's not perfect because as the target file gets edited that line number will change and the link will point to the wrong line.

For example if I point to BuJo.md:10 but then add a line at the beginning of the file that link would no longer point to the same text.

I found in the meantime that projects like logseq which use orgmode have properties for each line that allow to reference an id of the line instead of the line number. This would be more future proof as the wiki gets reviewed and changed.

**Comment by tinmarino** [comment]
I like the regex or line number style. In short this is more than enought. Have a look at the tag implementation which is well made.

Fir the id of each line, this ends reimplementing git. So if, optionally you can get a git reference to the "blob" the anchor just would have to be prefixed by git. This leads to dificulties in mantaining ones wiki.
